File: ClassLibrary1/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ClassLibrary1
{
    public class Class1
    {
        public const string DEFAULT_VALUE = "TEST";
        public string Property1 { get { return _field1; } }
        private string _field1;

        private void ResetProperty1()
        {
            throw new NotImplementedException();
        }

        public void Initialize()
        {
            ResetProperty1();
        }
    }
}

File: PHP/Class1.php
<?php
class Class1
{
    const DEFAULT_VALUE = "TEST";
    private $_field1;

    private function resetProperty1(){
        throw new Exception("resetProperty1() not implemented.");
    }

    public function getProperty1()
    {
        return $this->_field1;
    }
    public function initialize(){
        $this->resetProperty1();
    }
}